### DISCUSSION TITLE
[Feature] Original Image and Video Transcoding

### DISCUSSION THREAD

**Discussion by AngelaDMerkel** [immich-app/immich#8907] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Immich currently offers transcoding for serving and display to the user, but it would also be useful to offer the same functionality for replacing the originals. 

For example:

JPEG → Lossless JXL
h264 MP4 → h265/AV1 MP4

A lot of videos and images uploaded from mobile devices aren't as optimised as they could be and over time transcoding a replacement for storage could save the user, especially if they have other family members who take a lot of video. I know this might go against some of the principles of Immich as a backup tool, but I see no reason why the default assumption is that the used wants to store a 1GB a minute video taken in portrait of their meal from 6 years ago. A memory to savour sure, but hardly one which merits original quality. 

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by krkrs** [comment]
libjxl library can transcode JPEG to Lossless JXL and back to exact JPEG while keeping the original metadata. The server could transcode images to JXL on upload, and re-transcode to JPEG on download. This approach wont go against the backup principle. Also, user could download transcoded version. When it comes to video I have mixed feelings.

**Comment by bo0tzz** [comment]
#1202
